[addons] add addon filesystem callback functions to new style

AlwinEsch opened the pull request:
This is the next step to add addon callbacks to the new way.
After them are all parts who already in old libXBMC_addon.h included on
new interface headers.

As next comes the change of GUI callbacks.

<!--- Provide a general summary of your change in the Title above -->

## Description
<!--- Describe your change in detail -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

ksooo commented on xbmc/addons/interfaces/Filesystem.cpp:
```diff
@@ -0,0 +1,582 @@
+  if (addon == nullptr || file == nullptr)
+  {
+    CLog::Log(LOGERROR, "Interface_Filesystem::%s - invalid data (addon='%p', file='%p')", __FUNCTION__, addon, file);
+    return 0.0f;
```
This function returns an int. So this should read <code>return 0;</code> or even better <code>return -1;</code> as <code>CFile::Truncate</code> returns -1 in error case.

ksooo commented on xbmc/addons/interfaces/Filesystem.cpp:
```diff
@@ -0,0 +1,582 @@
+  if (addon == nullptr || file == nullptr)
+  {
+    CLog::Log(LOGERROR, "Interface_Filesystem::%s - invalid data (addon='%p', file='%p')", __FUNCTION__, addon, file);
+    return 0.0f;
```
This function returns an int64_t. So this should read <code>return 0;</code> or even better <code>return -1;</code> as <code>CFile::Seek</code> returns -1 in error case.

ksooo commented on xbmc/addons/interfaces/Filesystem.cpp:
```diff
@@ -0,0 +1,582 @@
+  if (addon == nullptr || file == nullptr)
+  {
+    CLog::Log(LOGERROR, "Interface_Filesystem::%s - invalid data (addon='%p', file='%p')", __FUNCTION__, addon, file);
+    return 0;
```
<code>return -1;</code>?

ksooo commented on xbmc/addons/interfaces/Filesystem.cpp:
```diff
@@ -0,0 +1,582 @@
+  if (addon == nullptr || file == nullptr)
+  {
+    CLog::Log(LOGERROR, "Interface_Filesystem::%s - invalid data (addon='%p', file='%p')", __FUNCTION__, addon, file);
+    return 0;
```
<code>return -1;</code>?

ksooo commented on xbmc/addons/kodi-addon-dev-kit/include/kodi/Filesystem.h:
```diff
@@ -0,0 +1,1486 @@
+    std::string m_label;
+    std::string m_title;
+    std::string m_path;
+    std::map<std::string, std::string> m_property;
```
should either read m_propertyMap or m_properties

ksooo commented on xbmc/addons/kodi-addon-dev-kit/include/kodi/Filesystem.h:
```diff
@@ -0,0 +1,1486 @@
+  /// @param[in] minimum The minimum size (or maybe the minimum number of chunks?)
+  /// @return The aligned size
+  ///
+  static inline unsigned int GetChunkSize(unsigned int chunk, unsigned int minimum)
```
Some of the global function in the header are static, others are not. What is the criteria? Should be consistent.

ksooo commented on xbmc/addons/kodi-addon-dev-kit/include/kodi/Filesystem.h:
```diff
@@ -0,0 +1,1486 @@
+    ssize_t Read(void* ptr, size_t size)
+    {
+      if (!m_file)
+        return 0;
```
<code>return -1;</code>?

ksooo commented on xbmc/addons/kodi-addon-dev-kit/include/kodi/Filesystem.h:
```diff
@@ -0,0 +1,1486 @@
+    ssize_t Write(const void* ptr, size_t size)
+    {
+      if (!m_file)
+        return 0;
```
<code>return -1;</code>?

ksooo commented on xbmc/addons/kodi-addon-dev-kit/include/kodi/Filesystem.h:
```diff
@@ -0,0 +1,1486 @@
+    int64_t Seek(int64_t position, int whence = SEEK_SET)
+    {
+      if (!m_file)
+        return 0;
```
<code>return -1;</code>?

ksooo commented on xbmc/addons/kodi-addon-dev-kit/include/kodi/Filesystem.h:
```diff
@@ -0,0 +1,1486 @@
+    int64_t GetLength()
+    {
+      if (!m_file)
+        return 0;
```
<code>return -1;</code>?

ksooo reviewed (changes requested):
Some suggestions and questions.

AlwinEsch commented on xbmc/addons/interfaces/Filesystem.cpp:
```diff
@@ -0,0 +1,592 @@
+  if (!entries)
+    return;
+
+  int toCopy = std::min(num_entries, static_cast<unsigned int>(items.Size()));
```
Becomes changed, was a bit a c&p https://github.com/xbmc/xbmc/blob/master/xbmc/addons/interfaces/Addon/AddonCallbacksAddon.cpp#L582-L598 :)

ksooo replied:
Okay.

AlwinEsch committed:
[addons] add addon filesystem callback functions to new style

This is the next step to add addon callbacks to the new way.
After them are all parts who already in old libXBMC_addon.h included on
new interface headers.

As next comes the change of GUI callbacks.

Because of the size is the doxygen part not included here and comes in another
commit.

AlwinEsch commented:
Is updated, thanks for review!

ksooo commented:
Okay to merge, imo.

AlwinEsch commented:
jenkins build this please
